sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_event.cpp

```diff
@@ -1001,6 +1001,75 @@ ur_result_t EventCreate(ur_context_handle_t Context, ur_queue_handle_t Queue,
   return UR_RESULT_SUCCESS;
 }
 
+// Helper function for creating a PI event with a specific EventType
+// The "Queue" argument specifies the PI queue where a command is submitted.
+// The "HostVisible" argument specifies if event needs to be allocated from
+// a host-visible pool.
+// The "EventType" argument specifies the event type that will be
+// associated to the created event.
+//
+ur_result_t EventCreateWithExplicitType(ur_context_handle_t Context,
```

Comment:
This is quite a lot of code duplication. I think we could just set the command type manually after calling `EventCreate` here instead of duplicating the entire function.